developer/error-handling-guide.md

```diff
@@ -0,0 +1,39 @@
+# Error handling guide
```

Comment:
Do you think this is a good title/filename for this topic? Wasn't sure if `Error messaging guide` would be better.

Reply:
Alternatively, all this info could go in the Code Style Guide at the end: https://docs.reactioncommerce.com/reaction-docs/master/styleguide#copy-conventions

Reply:
I think so. There's certainly a ton more you could say about "Error Handling"but this is a good start.